RFC: Unbreak torch.is_vulkan_available() on Mac

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #154926
* #154925
* #154924
* #154678
* #154677
* #154676
* __->__ #154675

I am a Vulkan noob, but this extension and flag seem to be necessary. See "Encounted VK_ERROR_INCOMPATIBLE_DRIVER" at https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Instance .

(For anyone trying to repro at home, I have the following homebrew packages installed, not all of which may be necessary: molten-vk, vulkan-headers, vulkan-loader, vulkan-tools, vulkan-utility-libraries. I also have VK_ICD_FILENAMES set to /opt/homebrew/etc/vulkan/icd.d/MoltenVK_icd.json, and I built PyTorch with USE_VULKAN=1. Making sure vkcube works helped me debug this setup.)

Pull-Request-resolved: https://github.com/pytorch/pytorch/pull/153776

swolchok committed:
RFC: Unbreak torch.is_vulkan_available() on Mac

I am a Vulkan noob, but this extension and flag seem to be necessary. See "Encounted VK_ERROR_INCOMPATIBLE_DRIVER" at https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Instance .

(For anyone trying to repro at home, I have the following homebrew packages installed, not all of which may be necessary: molten-vk, vulkan-headers, vulkan-loader, vulkan-tools, vulkan-utility-libraries. I also have VK_ICD_FILENAMES set to /opt/homebrew/etc/vulkan/icd.d/MoltenVK_icd.json, and I built PyTorch with USE_VULKAN=1. Making sure vkcube works helped me debug this setup.)

ghstack-source-id: 76b67e71a671d40117e73e8a6a0f27f8a1e1b595
Pull-Request-resolved: https://github.com/pytorch/pytorch/pull/153776

[ghstack-poisoned]

swolchok committed:
Update on "RFC: Unbreak torch.is_vulkan_available() on Mac"


I am a Vulkan noob, but this extension and flag seem to be necessary. See "Encounted VK_ERROR_INCOMPATIBLE_DRIVER" at https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Instance .

(For anyone trying to repro at home, I have the following homebrew packages installed, not all of which may be necessary: molten-vk, vulkan-headers, vulkan-loader, vulkan-tools, vulkan-utility-libraries. I also have VK_ICD_FILENAMES set to /opt/homebrew/etc/vulkan/icd.d/MoltenVK_icd.json, and I built PyTorch with USE_VULKAN=1. Making sure vkcube works helped me debug this setup.)

Pull-Request-resolved: https://github.com/pytorch/pytorch/pull/153776

[ghstack-poisoned]

malfet reviewed (approved):
I guess this is fine (I assume this codepath are never hit on iOS and we don't compile with Vulkan on Mac by default. so...)

swolchok committed:
Update on "RFC: Unbreak torch.is_vulkan_available() on Mac"


I am a Vulkan noob, but this extension and flag seem to be necessary. See "Encounted VK_ERROR_INCOMPATIBLE_DRIVER" at https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Instance .

(For anyone trying to repro at home, I have the following homebrew packages installed, not all of which may be necessary: molten-vk, vulkan-headers, vulkan-loader, vulkan-tools, vulkan-utility-libraries. I also have VK_ICD_FILENAMES set to /opt/homebrew/etc/vulkan/icd.d/MoltenVK_icd.json, and I built PyTorch with USE_VULKAN=1. Making sure vkcube works helped me debug this setup.)

Pull-Request-resolved: https://github.com/pytorch/pytorch/pull/153776

[ghstack-poisoned]

swolchok commented:
> If you think this is a mistake, please feel free to reopen and contact Dev Infra.

done

swolchok committed:
Update on "RFC: Unbreak torch.is_vulkan_available() on Mac"


I am a Vulkan noob, but this extension and flag seem to be necessary. See "Encounted VK_ERROR_INCOMPATIBLE_DRIVER" at https://vulkan-tutorial.com/Drawing_a_triangle/Setup/Instance .

(For anyone trying to repro at home, I have the following homebrew packages installed, not all of which may be necessary: molten-vk, vulkan-headers, vulkan-loader, vulkan-tools, vulkan-utility-libraries. I also have VK_ICD_FILENAMES set to /opt/homebrew/etc/vulkan/icd.d/MoltenVK_icd.json, and I built PyTorch with USE_VULKAN=1. Making sure vkcube works helped me debug this setup.)

Pull-Request-resolved: https://github.com/pytorch/pytorch/pull/153776

[ghstack-poisoned]

swolchok commented:
sent a new PR #155595 due to merge issues
